scipy/integrate/quadrature.py

```diff
@@ -396,35 +417,102 @@ def simps(y, x=None, dx=1, axis=-1, even='avg'):
         if x.shape[axis] != N:
             raise ValueError("If given, length of x along axis must be the "
                     "same as y.")
-    if N % 2 == 0:
-        val = 0.0
-        result = 0.0
-        slice1 = (slice(None),)*nd
-        slice2 = (slice(None),)*nd
-        if even not in ['avg', 'last', 'first']:
-            raise ValueError("Parameter 'even' must be 'avg', 'last', or 'first'.")
-        # Compute using Simpson's rule on first intervals
-        if even in ['avg', 'first']:
-            slice1 = tupleset(slice1, axis, -1)
-            slice2 = tupleset(slice2, axis, -2)
-            if x is not None:
-                last_dx = x[slice1] - x[slice2]
-            val += 0.5*last_dx*(y[slice1]+y[slice2])
-            result = _basic_simps(y,0,N-3,x,dx,axis)
-        # Compute using Simpson's rule on last set of intervals
-        if even in ['avg', 'last']:
-            slice1 = tupleset(slice1, axis, 0)
-            slice2 = tupleset(slice2, axis, 1)
-            if x is not None:
-                first_dx = x[tuple(slice2)] - x[tuple(slice1)]
-            val += 0.5*first_dx*(y[slice2]+y[slice1])
-            result += _basic_simps(y,1,N-2,x,dx,axis)
-        if even == 'avg':
-            val /= 2.0
-            result /= 2.0
-        result = result + val
+        # checking whether x is equally spaced across all rows.
+        elif y.shape == x.shape and method in ['second', 'extended']:
+            h = diff(x, n=2)
+            if not np.all(h == 0):
+                    raise ValueError("If given, and method used is extended or second " 
+                        "x must be equally spaced everywhere.")
+
+    # Implementing Simpson's 3/8-th rule.
+    
+    if method is 'second' and (n % 3) != 0:
+        raise ValueError("If using Simpson's second rule, number of samples "
+            "must be of the form 3*k+1.")
```

Comment:
Ugh. I see, https://github.com/scipy/scipy/pull/4406/files#r23496962
Please don't mix uppercase-en and lowercase-en unless it's really necessary. 
Does the variable need to be lowercase-en?